test/models/concerns/key_pair_compute_resource_test.rb

```diff
@@ -0,0 +1,49 @@
+require 'test_helper'
+
+class DummyComputeResource < ComputeResource
+  include KeyPairComputeResource
+
+  def client
+    @client ||= ::Fog::Compute.new(:provider => 'AWS', :aws_access_key_id => '123', :aws_secret_access_key => 'abcd')
+  end
+end
+
+class KeyPairComputeResourceTest < ActiveSupport::TestCase
+  setup { Fog.mock! }
+  teardown { Fog.unmock! }
+
+  test 'should remove the key pair on compute resource deletion' do
+    cr = DummyComputeResource.new
+    key_pair = FactoryGirl.build(:key_pair)
```

Comment:
This should also have `.expects(:destroy)`, though it also seems to be destroyed twice (in a quick test) which needs checking. Otherwise nothing asserts that the key is destroyed on deletion.